test/rpc-server/implies/implied-substitution/response.golden

```diff
@@ -0,0 +1 @@
+{"result":{"implication":{"format":"KORE","term":{"tag":"Implies","first":{"tag":"EVar","name":"a","sort":{"args":[],"tag":"SortApp","name":"SortK"}},"second":{"tag":"Exists","var":"x","arg":{"tag":"EVar","name":"x","sort":{"args":[],"tag":"SortApp","name":"SortK"}},"varSort":{"args":[],"tag":"SortApp","name":"SortK"},"sort":{"args":[],"tag":"SortApp","name":"SortK"}},"sort":{"args":[],"tag":"SortApp","name":"SortK"}},"version":1},"satisfiable":true,"condition":{"substitution":{"format":"KORE","term":{"tag":"Equals","first":{"tag":"EVar","name":"a","sort":{"args":[],"tag":"SortApp","name":"SortK"}},"second":{"tag":"EVar","name":"x","sort":{"args":[],"tag":"SortApp","name":"SortK"}},"sort":{"tag":"SortVar","name":"JSONSortVariable"},"argSort":{"args":[],"tag":"SortApp","name":"SortK"}},"version":1},"predicate":{"format":"KORE","term":{"tag":"Top","sort":{"args":[],"tag":"SortApp","name":"SortK"}},"version":1}}},"jsonrpc":"2.0","id":1}
```

Comment:
If I'm reading this right, the substitution looks like `a #Equals x`.

This is out of the scope of this PR, but we should make sure that the substitutions' order is preserved. One situation where this matters is when you have `var1 #Equals var2`, the way you order that will give different results.

Please open a subsequent PR when you have a chance which fixes this.

Reply:
> If I'm reading this right, the substitution looks like `a #Equals x`.

That's right.
 
> This is out of the scope of this PR, but we should make sure that the substitutions' order is preserved. One situation where this matters is when you have `Var1 #Equals Var2`, the way you order that will give different results.

I hope the order is indeed stable. The code that translates substitutions to json is `Kore.Syntax.Json.fromSubstitution`, and goes left-to-right (`foldl1`) over `Substitution.unwrap`
https://github.com/runtimeverification/haskell-backend/blob/master/kore/src/Kore/Syntax/Json.hs#L115
 
> Please open a subsequent PR when you have a chance which fixes this.